uikit/src/main/java/com/midtrans/sdk/uikit/views/banktransfer/status/BankTransferStatusPresenter.java

```diff
@@ -111,14 +113,66 @@ public String getMandiriBillExpiration() {
     }
 
     public String getBankCode() {
-        String bankCode = LABEL_BANK_CODE_BNI;
 
-        MerchantPreferences preferences = getMidtransSDK().getMerchantData().getPreference();
-        if (preferences != null && !TextUtils.isEmpty(preferences.getOtherVaProcessor())
-                && preferences.getOtherVaProcessor().equals(UiKitConstants.OTHER_VA_PROCESSOR_PERMATA)) {
-            bankCode = LABEL_BANK_CODE_PERMATA;
+        String bankCode = "";
+        String bank = getOtherVaProcessor();
+
+        switch (bank) {
+            case BRI:
+                bankCode = LABEL_BANK_CODE_BRI;
+                break;
+            case BNI:
+                bankCode = LABEL_BANK_CODE_BNI;
+                break;
+            case PERMATA:
+                bankCode = LABEL_BANK_CODE_PERMATA;
+                break;
         }
 
         return bankCode;
     }
+
+    private String getOtherBankVAExpiration() {
```

Comment:
Hi. I see this code may have different outcome than your previous implementation.

Reply:
equivalent result should looks like this:
`String expiration = "";
        String otherVaProcessor = getOtherVaProcessor();

        switch (otherVaProcessor) {
            case BRI:
                expiration = transactionResponse.getBriExpiration();
                break;
            case BNI:
                expiration = transactionResponse.getBniExpiration();
                break;
            default:
                expiration = transactionResponse.getPermataExpiration();
                break;
        }

        return expiration;`